cql3/user_types.cc

```diff
@@ -295,4 +295,25 @@ void user_types::setter::execute(mutation& m, const clustering_key_prefix& row_k
     }
 }
 
+void user_types::setter_by_field::execute(mutation& m, const clustering_key_prefix& row_key, const update_parameters& params) {
+    assert(column.type->is_user_type() && column.type->is_multi_cell());
+
+    auto value = _t->bind(params._options);
+    if (value == constants::UNSET_VALUE) {
+        return;
+    }
+
+    auto& type = static_cast<const user_type_impl&>(*column.type);
+
+    auto idx = type.idx_of_field(_field_name);
+    assert(idx && *idx < type.size());
+
```

Comment:
Shoud we not have computed idx during prepare?

This assert() is dangeous, if we made a mistake then this can crash the server. As long as we capture the type during prepare everything should be self-consistent. Better to throw something.

Reply:
I'm not sure about our policy with assertions versus throws for bugs.
1. An assertion is (?) better than segfaulting by dereferencing a null pointer or even worse, going into undefined behaviour by accessing an array out of bounds or something,
2. But then do we want to crash the server in case like that? Or maybe we should throw there saying "we no longer go into this territory due to a bug, but we can still run the rest of the code"?

Maybe it should depend on the layer? Like when something is wrong during the CQL phase we can throw so only some features will be disabled due to a bug, but if something is wrong deeper in the code then it just doesn't make sense to keep the DB running anymore?

In any case, I'll see if I can move idx to the prepare phase. Indeed it looks like I don't need `_field_name` anywhere else.

Reply:
Moved `idx` to `prepare`.

Reply:
Yes, in the front-end throwing is often better than assersions. In the back end usually one can't recover so it's better to fail fast.